packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,453 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title The Nouns art storage contract
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+import { INounsArt } from './interfaces/INounsArt.sol';
+import { Inflate } from './libs/Inflate.sol';
+import { SSTORE2 } from './libs/SSTORE2.sol';
+
+contract NounsArt is INounsArt {
+    // Current Nouns Descriptor address
+    address public override descriptor;
+
+    // Pending Nouns Descriptor address
+    address public override pendingDescriptor;
+
+    // Noun Backgrounds (Hex Colors)
+    string[] public override backgrounds;
+
+    // Noun Color Palettes (Index => Hex Colors)
+    mapping(uint8 => address) public palettesPointers;
+
+    // Noun Bodies (Custom RLE)
+    Trait public _bodies;
+
+    // Noun Accessories (Custom RLE)
+    Trait public _accessories;
+
+    // Noun Heads (Custom RLE)
+    Trait public _heads;
+
+    // Noun Glasses (Custom RLE)
+    Trait public _glasses;
+
+    /**
+     * @notice Require that the sender is the descriptor.
+     */
+    modifier onlyDescriptor() {
+        if (msg.sender != descriptor) {
+            revert SenderIsNotDescriptor();
+        }
+        _;
+    }
+
+    constructor(address _descriptor) {
+        descriptor = _descriptor;
+    }
+
+    /**
+     * @notice Set the pending descriptor, which can be confirmed
+     * by calling `confirmDescriptor`.
+     * @dev This function can only be called by the current descriptor.
+     */
+    function setDescriptor(address _pendingDescriptor) external override onlyDescriptor {
+        pendingDescriptor = _pendingDescriptor;
+    }
+
+    /**
+     * @notice Confirm the pending descriptor.
+     * @dev This function can only be called by the pending descriptor.
+     */
+    function confirmDescriptor() external override {
+        if (msg.sender != pendingDescriptor) {
+            revert SenderIsNotPendingDescriptor();
+        }
+
+        address oldDescriptor = descriptor;
+        descriptor = pendingDescriptor;
+        delete pendingDescriptor;
+
+        emit DescriptorUpdated(oldDescriptor, descriptor);
+    }
+
+    /**
+     * @notice Get the Trait struct for bodies.
+     * @return Trait the struct, including a total image count, and an array of storage pages.
+     */
+    function bodiesTrait() external view override returns (Trait memory) {
```

Comment:
we can use the default getter for the variable, and move the natspec to the variable definition

Reply:
➕